source/sass/mixins.scss

```diff
@@ -118,76 +133,50 @@
   padding-top: calc(#{$offset} + #{$extra-vertical-spacing});
 }
 
-// TODO: override Bootstrap's "@mixin button-outline-variant" so we can reduce our overrides for buttons
-@mixin button-basic-style(
-  $default,
-  $default-hover,
-  $dark-theme-default,
-  $dark-theme-hover
-) {
-  color: map-get($default, "color");
-  background-color: map-get($default, "background-color");
-  border-color: map-get($default, "border-color");
+@mixin button-base-colors($styles) {
+  color: map-get($styles, "color");
+  background-color: map-get($styles, "background-color");
+  border-color: map-get($styles, "border-color");
+}
 
-  &.active {
-    color: map-get($default-hover, "color");
-    background-color: map-get($default-hover, "background-color");
-    border-color: map-get($default-hover, "border-color");
+@mixin button-colors-by-mode($mode-colors) {
+  $default-colors: map-get($mode-colors, "default");
+  $hover-colors: map-get($mode-colors, "hover-focus");
+  $active-colors: map-get($mode-colors, "active");
+
+  @include button-base-colors($default-colors);
+
+  @include hover-focus {
+    @include button-base-colors($hover-colors);
   }
 
-  @include hover-focus-active {
-    @extend .active;
+  @include active {
+    @include button-base-colors($active-colors);
   }
 
-  &.disabled,
-  &:disabled,
-  &[disabled] {
+  @include disabled {
+    @include button-base-colors($default-colors);
+  }
+
+  &:not(:disabled):not(.disabled) {
```

Comment:
Does `&:not(:disabled, .disabled)` work? Or does each thing need to be wrapped in its own `not()`?

Reply:
I think `:not()` only allow one argument.